### ISSUE TITLE
Querying JSON array returns null

### ISSUE THREAD

**Issue by LeonRyan** [nuxt/content#1814]
### Environment

------------------------------
- Operating System: `Linux`
- Node Version:     `v18.12.0`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@8.19.2`
- Builder:          `vite`
- User Config:      `components`, `imports`, `alias`, `css`, `postcss`, `runtimeConfig`, `modules`, `content`, `build`, `tailwindcss`, `ignore`, `routeRules`, `hooks`, `sitemap`
- Runtime Modules:  `@pinia/nuxt@0.4.6`, `@vueuse/nuxt@9.10.0`, `nuxt-lodash@2.4.1`, `@nuxt/content@2.3.0`, `@nuxtjs/tailwindcss@6.2.0`
- Build Modules:    `-`
------------------------------

### Reproduction

Run `npm run generate` on the following Stackblitz to see the comparison.

Nuxt Content V1 (Works):
https://stackblitz.com/edit/nuxt-starter-aidydt?file=pages/index.vue

Nuxt Content V2 (Doesn't work):
https://stackblitz.com/edit/github-yyzyb8?file=app.vue

### Describe the bug

In Nuxt Content V1 we could query JSON arrays, however in V2 it returns null.

I think this might be because Nitro doesn't support JSON arrays and them into a body key, so a JSON file that was `[...data]` becomes `{ body: ...data }`, as you can see from the console warning:
``` WARN  JSON array is not supported in content:data.json, moving the array into the body key```

Is there a way around this? The Nuxt Content JSON documentation is pretty bare bones. I've tried creating a hook to flatten the JSON but it doesn't work.

**Comment by Barbapapazes** [comment]
Yes, [missing support for arrays](https://content.nuxtjs.org/v1/getting-started/writing#arrays) and I think (I will verify) that previously, we were able to filter on data created by an array using a `where`. Now, we need to load the entier JSON file and then to search. 

It could be a gread addition if we were able to create a filter request to search inside a file and return the correct result.
